### PR TITLE
Now showing stripe connect and webhook settings for selected environment

### PR DESCRIPTION
**Pull request by dparker1005**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:
- Only show Stripe Connect button for selected gateway environment
- Only show Stripe webhook status for selected gateway environment
- Added new html class structure on payment settings page to allow showing/hiding fields only for particular gateway environment (gateway_[gateway_name]_[gateway_environment])

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: adminpages/paymentsettings.php [hunk 1]
```diff
@@ -127,7 +127,7 @@
 					<label for="gateway"><?php _e('Payment Gateway', 'paid-memberships-pro' );?>:</label>
 				</th>
 				<td>
-					<select id="gateway" name="gateway" onchange="pmpro_changeGateway(jQuery(this).val());">
+					<select id="gateway" name="gateway" onchange="pmpro_changeGateway(jQuery('#gateway').val(), jQuery('#gateway_environment').val());">
```
Comment (sc0ttkclark) [inline adminpages/paymentsettings.php:130]: Why pass any arguments in at all if we can always just get the selects targeted by the ID?

Comment (sc0ttkclark) [reply adminpages/paymentsettings.php:130]:
```suggestion
					<select id="gateway" name="gateway">
```

File: adminpages/paymentsettings.php [hunk 2]
```diff
@@ -148,16 +148,17 @@
 					<label for="gateway_environment"><?php _e('Gateway Environment', 'paid-memberships-pro' );?>:</label>
 				</th>
 				<td>
-					<select name="gateway_environment">
+					<select id="gateway_environment" name="gateway_environment" onchange="pmpro_changeGateway(jQuery('#gateway').val(), jQuery('#gateway_environment').val());">
```
Comment (sc0ttkclark) [inline adminpages/paymentsettings.php:151]:
```suggestion
					<select id="gateway_environment" name="gateway_environment">
```

File: adminpages/paymentsettings.php [hunk 3]
```diff
@@ -148,16 +148,17 @@
...
 						<option value="sandbox" <?php selected( $gateway_environment, "sandbox" ); ?>><?php _e('Sandbox/Testing', 'paid-memberships-pro' );?></option>
 						<option value="live" <?php selected( $gateway_environment, "live" ); ?>><?php _e('Live/Production', 'paid-memberships-pro' );?></option>
 					</select>
 					<script>
-						function pmpro_changeGateway(gateway)
+						function pmpro_changeGateway(gateway, gateway_environment)
```
Comment (sc0ttkclark) [inline adminpages/paymentsettings.php:156]:
```suggestion
						function pmpro_changeGateway()
```

File: adminpages/paymentsettings.php [hunk 4]
```diff
@@ -148,16 +148,17 @@
...
 						{
 							//hide all gateway options
```
Comment (sc0ttkclark) [inline adminpages/paymentsettings.php:158]:
```suggestion
							const gateway = jQuery('#gateway').val();
							const gateway_environment = jQuery('#gateway_environment').val();

							//hide all gateway options
```

File: adminpages/paymentsettings.php [hunk 5]
```diff
@@ -173,7 +174,7 @@ function pmpro_changeGateway(gateway)
 								jQuery('#pmpro-default-gateway-message').hide();
 							}
 						}
-						pmpro_changeGateway(jQuery('#gateway').val());
+						pmpro_changeGateway(jQuery('#gateway').val(), jQuery('#gateway_environment').val());
```
Comment (sc0ttkclark) [inline adminpages/paymentsettings.php:177]:
```suggestion
						pmpro_changeGateway();

						// Handle change events.
						jQuery('#gateway, #gateway_environment').on( 'change', 'pmpro_changeGateway' );
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 6]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
-        <tr class="gateway gateway_stripe" <?php if ( $gateway != "stripe" ) { ?>style="display: none;"<?php } ?>>
+        <tr class="gateway <?php if ( ! self::using_legacy_keys() ) { echo 'gateway_stripe_live'; } ?>" <?php if ( $gateway != "stripe" || $gateway_environment != "live" || self::using_legacy_keys() ) { ?>style="display: none;"<?php } ?>>
+            <th scope="row" valign="top">
+                <label><?php _e( 'Webhook (Live)', 'paid-memberships-pro' ); ?>:</label>
+            </th>
+            <td>
+				<?php
+					$last_webhook = get_option( 'pmpro_stripe_last_webhook_recieved_live' );
+					if ( ! empty( $last_webhook ) ) {
+						echo '<p>' . esc_html( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:463]:
```suggestion
						echo '<p>' . esc_html__( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 7]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+					} else {
+						echo '<p>' . esc_html( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:465]:
```suggestion
						echo '<p>' . esc_html__( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 8]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+					}
+					if ( ! self::webhook_is_working( 'live' ) ) {
+						echo '<p class="pmpro_error">' . esc_html( 'Your webhook may not be working correctly.', 'paid-memberships-pro' ) . '</p>';
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:468]:
```suggestion
						echo '<p class="pmpro_error">' . esc_html__( 'Your webhook may not be working correctly.', 'paid-memberships-pro' ) . '</p>';
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 9]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+					}
+				?>
+				<p class="description"><?php esc_html_e( 'Webhook URL', 'paid-memberships-pro' ); ?>:
+				<code><?php echo self::get_site_webhook_url(); ?></code></p>
+            </td>
+        </tr>
+		<tr class="gateway <?php if ( ! self::using_legacy_keys() ) { echo 'gateway_stripe_sandbox'; } ?>" <?php if ( $gateway != "stripe" || $gateway_environment != "sandbox" || self::using_legacy_keys() ) { ?>style="display: none;"<?php } ?>>
+            <th scope="row" valign="top">
+                <label><?php _e( 'Webhook (Sandbox)', 'paid-memberships-pro' ); ?>:</label>
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:477]:
```suggestion
                <label><?php esc_html_e( 'Webhook (Sandbox)', 'paid-memberships-pro' ); ?>:</label>
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 10]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+					}
+				?>
+				<p class="description"><?php esc_html_e( 'Webhook URL', 'paid-memberships-pro' ); ?>:
+				<code><?php echo self::get_site_webhook_url(); ?></code></p>
+            </td>
+        </tr>
+		<tr class="gateway <?php if ( self::using_legacy_keys() ) { echo 'gateway_stripe'; } ?>" <?php if ( $gateway != "stripe" || ! self::using_legacy_keys() ) { ?>style="display: none;"<?php } ?>>
             <th scope="row" valign="top">
-                <label><?php _e( 'Webhook', 'paid-memberships-pro' ); ?>:</label>
+                <label><?php _e( 'Webhook (Legacy)', 'paid-memberships-pro' ); ?>:</label>
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:497]:
```suggestion
                <label><?php esc_html_e( 'Webhook (Legacy)', 'paid-memberships-pro' ); ?>:</label>
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 11]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+        </tr>
+		<tr class="gateway <?php if ( ! self::using_legacy_keys() ) { echo 'gateway_stripe_sandbox'; } ?>" <?php if ( $gateway != "stripe" || $gateway_environment != "sandbox" || self::using_legacy_keys() ) { ?>style="display: none;"<?php } ?>>
+            <th scope="row" valign="top">
+                <label><?php _e( 'Webhook (Sandbox)', 'paid-memberships-pro' ); ?>:</label>
+            </th>
+            <td>
+				<?php
+					$last_webhook = get_option( 'pmpro_stripe_last_webhook_recieved_sandbox' );
+					if ( ! empty( $last_webhook ) ) {
+						echo '<p>' . esc_html( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:483]:
```suggestion
						echo '<p>' . esc_html__( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 12]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+					} else {
+						echo '<p>' . esc_html( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
+					}
+					if ( ! self::webhook_is_working( 'sandbox' ) ) {
+						echo '<p class="pmpro_error">' . esc_html( 'Your webhook may not be working correctly.', 'paid-memberships-pro' ) . '</p>';
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:488]:
```suggestion
						echo '<p class="pmpro_error">' . esc_html__( 'Your webhook may not be working correctly.', 'paid-memberships-pro' ) . '</p>';
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 13]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+            <th scope="row" valign="top">
+                <label><?php _e( 'Webhook (Sandbox)', 'paid-memberships-pro' ); ?>:</label>
+            </th>
+            <td>
+				<?php
+					$last_webhook = get_option( 'pmpro_stripe_last_webhook_recieved_sandbox' );
+					if ( ! empty( $last_webhook ) ) {
+						echo '<p>' . esc_html( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
+					} else {
+						echo '<p>' . esc_html( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:485]:
```suggestion
						echo '<p>' . esc_html__( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 14]
```diff
@@ -3599,6 +3638,7 @@ public static function webhook_is_working() {
 			SELECT COUNT(*)
 			FROM $wpdb->pmpro_membership_orders
 			WHERE gateway = 'stripe'
+			AND gateway_environment = '" . $gateway_environment . "'
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:3641]: This and the other vars should be passed in via `%s` and a `$wpdb->prepare()` call

File: classes/gateways/class.pmprogateway_stripe.php [hunk 15]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+						echo '<p>' . esc_html( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
+					} else {
+						echo '<p>' . esc_html( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
+					}
+					if ( ! self::webhook_is_working( 'sandbox' ) ) {
+						echo '<p class="pmpro_error">' . esc_html( 'Your webhook may not be working correctly.', 'paid-memberships-pro' ) . '</p>';
+					}
+				?>
+				<p class="description"><?php esc_html_e( 'Webhook URL', 'paid-memberships-pro' ); ?>:
+				<code><?php echo self::get_site_webhook_url(); ?></code></p>
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:492]:
```suggestion
				<code><?php echo esc_html( self::get_site_webhook_url() ); ?></code></p>
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 16]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
+						echo '<p>' . esc_html( 'Last webhook recieved at', 'paid-memberships-pro' ) . ': ' . esc_html( $last_webhook ) . ' GMT.</p>';
+					} else {
+						echo '<p>' . esc_html( 'No webhooks have been recieved.', 'paid-memberships-pro' ) . '</p>';
+					}
+					if ( ! self::webhook_is_working( 'live' ) ) {
+						echo '<p class="pmpro_error">' . esc_html( 'Your webhook may not be working correctly.', 'paid-memberships-pro' ) . '</p>';
+					}
+				?>
+				<p class="description"><?php esc_html_e( 'Webhook URL', 'paid-memberships-pro' ); ?>:
+				<code><?php echo self::get_site_webhook_url(); ?></code></p>
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:472]:
```suggestion
				<code><?php echo esc_html( self::get_site_webhook_url() ); ?></code></p>
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 17]
```diff
@@ -443,17 +444,57 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
...
                 <label for="stripe_secretkey"><?php _e( 'Legacy Secret Key', 'paid-memberships-pro' ); ?>:</label>
             </th>
             <td>
                 <input type="text" id="stripe_secretkey" name="stripe_secretkey" value="<?php echo esc_attr( $values['stripe_secretkey'] ) ?>" autocomplete="off" class="regular-text code pmpro-admin-secure-key" />
             </td>
         </tr>
-        <tr class="gateway gateway_stripe" <?php if ( $gateway != "stripe" ) { ?>style="display: none;"<?php } ?>>
+        <tr class="gateway <?php if ( ! self::using_legacy_keys() ) { echo 'gateway_stripe_live'; } ?>" <?php if ( $gateway != "stripe" || $gateway_environment != "live" || self::using_legacy_keys() ) { ?>style="display: none;"<?php } ?>>
+            <th scope="row" valign="top">
+                <label><?php _e( 'Webhook (Live)', 'paid-memberships-pro' ); ?>:</label>
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:457]:
```suggestion
                <label><?php esc_html_e( 'Webhook (Live)', 'paid-memberships-pro' ); ?>:</label>
```

File: classes/gateways/class.pmprogateway_stripe.php [hunk 18]
```diff
@@ -338,7 +339,7 @@ static function pmpro_payment_option_fields( $values, $gateway ) {
 				<h2><?php _e( 'Stripe Settings', 'paid-memberships-pro' ); ?></h2>
```
Comment (sc0ttkclark) [inline classes/gateways/class.pmprogateway_stripe.php:339]:
```suggestion
				<h2><?php esc_html_e( 'Stripe Settings', 'paid-memberships-pro' ); ?></h2>
```

File: classes/gateways/class.pmprogateway_stripe.php (hunk 14 above)
Comment (dparker1005) [reply classes/gateways/class.pmprogateway_stripe.php]: Fixed in commit https://github.com/strangerstudios/paid-memberships-pro/pull/1682/commits/d2055a05213a0f8e434a783f5a388b39921fa2ea

**Pull request merged**